### PR TITLE
Add enhancements implemented for my diploma thesis

### PR DESCRIPTION
**Pull request by zyzzyxdonta**
- write results to backup file
- add Score-P user regions
- support for ESSL and ArmPL
- methods for cache flushing
- GitLab CI (job for rocfft hidden because it fails due to this problem: https://github.com/RadeonOpenCompute/rocminfo/issues/38)

### CODE REVIEW

**Comment by psteinb** [comment]
Thanks for providing this code back to the public. You worked hard and brought some nice advancements to gearshifft. I need to think about making a new release. Please let me know when you are ready for review/merge.

**Comment by zyzzyxdonta** [comment]
@psteinb ready for review!

File: README.md [hunk 1]
```diff
@@ -164,6 +164,98 @@ make install
...
+|           94819 |  |  |  |  * instance=1
+                       ...
+|          528948 |  |  |  * transform_backward
+|           44079 |  |  |  |  * instance=1
+                       ...
+```
+
+This report states that e.g. every executed forward transformation (`transform_forward`) consumed 94,819 single precision
+floating point operations (Flops).
+For the sake of precise time measurement, this also includes one operation that is caused by the
```
Comment (psteinb) [inline README.md]: I don't see this `one operation` in the example above. Maybe you want to comment on the fact if this is depicted or not.

File: README.md [hunk 2]
```diff
@@ -164,6 +164,98 @@ make install
 
 </details>
```
Comment (psteinb) [inline README.md:175]: More a general term, the support for ESSL and ArmPL is not documented in the README file. We should fix that I think.

**Review by psteinb — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (zyzzyxdonta) [reply README.md:175]: There is no "Supported Libraries" section in general. The requirements section names clfft, cufft and fftw, rocfft is only mentioned in the issues, mkl isn't found anywhere.

**Comment by psteinb** [comment]
Then we should find a good spot somewhere. [This](https://github.com/mpicbg-scicomp/gearshifft#requirements) is semantically
related and hit by any reader early on. I think this list should go here.

**Comment by zyzzyxdonta** [comment]
I added some info https://github.com/mpicbg-scicomp/gearshifft/pull/151/commits/aa77c5c15e32b6e28d848519f567ec1dd98063b9

File: README.md [hunk 3]
```diff
@@ -14,12 +14,21 @@ If you want to just browse our results, see the [raw benchmark data](https://www
...
+  - FFTW 3.3.4+
+  - Intel Math Kernel Library (MKL)¹
+  - IBM Engineering and Scientific Subroutine Library (ESSL)¹
+  - ARM Performance Libraries (ArmPL)¹
 - Boost version 1.59+
   - should be compiled with same compiler version or ...
   - ... disable the C++11 ABI for GCC with the `-DGEARSHIFFT_CXX11_ABI=OFF` cmake option
 - [half-code](http://half.sourceforge.net) by [Christian Rau](http://sourceforge.net/users/rauy) for float16 support (currently used for cufft half precision FFTs)
 
+¹) The respective FFTW-style wrapper library is needed.
```
Comment (psteinb) [inline README.md]:
```suggestion
¹) The respective FFTW-style wrapper library is required for the integration in gearshifft.
```

**Comment by psteinb** [comment]
awesome, thanks!

File: README.md (hunk 1 above)
Comment (zyzzyxdonta) [reply README.md]: Does this make it more clear? https://github.com/mpicbg-scicomp/gearshifft/pull/151/commits/940ac73c28613a765b6d19c12f28cff68cf105ef

**Comment by psteinb** [comment]
Perfect!

**Review by psteinb — APPROVED** [review]
LGTM! Thank you again for this mindblowing contribution.

**Pull request merged**